### PR TITLE
HPCC-25632 refactor examples/efs Helm chart with storage planes

### PR DESCRIPTION
**Pull request by xwang2713**
Add support for sasha storage.
Add support for AWS EFS CSI driver.

There is extra step to setup CSI driver and related EFS security group configuration with EKS cluster with AWS CLI
It is done by running a bash script: install-efs-provisioner.sh.
The EFS Helm chart should work with/without AWS EFS CSI driver even though with CSI driver is recommended.
Tested with AWS EKS on region us-east-2

Assigned off by Xiaoming Wang <xiaoming.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

AWS EKS region us-east-2

### CODE REVIEW

**Comment by xwang2713** [comment]
I add Greg to review README.md. He actually helped to prepare the file.

File: helm/examples/efs/hpcc-efs/templates/NOTES.txt
Comment (jakesmith) [inline helm/examples/efs/hpcc-efs/templates/NOTES.txt:22]: Similar issues to : https://github.com/hpcc-systems/HPCC-Platform/pull/14797
Will hold off from reviewing further until gh-14797 is resovled/merged.

File: helm/examples/efs/README.md
Comment (jakesmith) [inline helm/examples/efs/README.md:53]: typo: yamls

File: helm/examples/efs/README.md
Comment (jakesmith) [inline helm/examples/efs/README.md:26]: The README should still at least mention (as with other storage examples), the fact that the values-retained... is just an example of what would be generated as output from installing hpcc-efs.
i.e. if users want to use a different name in place of 'awsstorage' they can, the output is generated and will be correct, where as the example which is hard-wired to 'awsstorage' will not.

File: helm/examples/efs/hpcc-efs/values.yaml
Comment (jakesmith) [inline helm/examples/efs/hpcc-efs/values.yaml:1]: hpcc-efs

File: helm/examples/efs/values-auto-efs.yaml
Comment (jakesmith) [inline helm/examples/efs/values-auto-efs.yaml:1]: azure->efs

File: helm/examples/efs/values-retained-efs.yaml
Comment (jakesmith) [inline helm/examples/efs/values-retained-efs.yaml:1]: azure->efs

**Review by jakesmith — COMMENTED** [review]
@xwang2713 - a few minor issues/comments only

File: helm/examples/efs/README.md
Comment (xwang2713) [reply helm/examples/efs/README.md:53]: Fixed

File: helm/examples/efs/README.md
Comment (xwang2713) [reply helm/examples/efs/README.md:26]: Add following: 
An example values file to be supplied when installing the HPCC chart.
NB: Either use the output auto-generated when installing the "hpcc-efs" helm chart, or ensure the names in your values files for the storage types match the PVC names created. "values-retained-efs.yaml" expects that helm chart installation name is "awsstorage". Change the PVC name accordingly if another name is used.

Comment (jakesmith) [reply helm/examples/efs/README.md:26]: sounds fine.

File: helm/examples/efs/hpcc-efs/values.yaml
Comment (xwang2713) [reply helm/examples/efs/hpcc-efs/values.yaml:1]: fixed

File: helm/examples/efs/values-auto-efs.yaml
Comment (xwang2713) [reply helm/examples/efs/values-auto-efs.yaml:1]: done

File: helm/examples/efs/values-retained-efs.yaml
Comment (xwang2713) [reply helm/examples/efs/values-retained-efs.yaml:1]: fixed

**Comment by xwang2713** [comment]
@g-pan please review examples/efs/README.md

**Review by g-pan — APPROVED** [review]
This looks ok

**Review by g-pan — APPROVED** [review]
approved

**Review by jakesmith — APPROVED** [review]
@xwang2713 - looks good, please squash

**Pull request merged**
